Framework/PythonInterface/plugins/algorithms/ComputeCalibrationCoefVan.py

```diff
@@ -0,0 +1,168 @@
+from mantid.api import PythonAlgorithm, AlgorithmFactory, WorkspaceProperty, Progress, InstrumentValidator
+from mantid.kernel import Direction
+import mantid.simpleapi as api
+import numpy as np
+from scipy import integrate
+import scipy as sp
+import mlzutils
```

Comment:
To avoid having Algorithms too intertwined with Mantid's persistent storage mechanism (The AnalysisDataService, better known as workspace list: Algorithms should never directly access any workspace information through the AnalysisDataService, and should never modify AnalysisDataService directly. This kind of direct access bypasses workspace read-write locking, can lead to race conditions between algorithms, and will make it very hard in the future to have session-based AnalysisDataServices. Seeing some of the use cases for mlzutils such as [DNSDetEffCorrVana](https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/plugins/algorithms/DNSDetEffCorrVana.py#L96:L129) I suggest all operations in the mlzutils that deal with clearing and querying workspace lists are refactored away and removed.

Some of the functions you have in the mlzutils look genuinely useful elsewhere `compare_properties` and `compare_mandatory` are related and to do with property comparison. So either, create a module called PropertyComparitor which your algorithm can use, or create a base class derived from `PythonAlgorithm`, that future and existing `Algorithms` can derive from. `do_gaussian_fit` only seems to be used in this *ComputeCalibrationCoefVan.py* at the moment, so I suggest you add it as a member function to that Algorithm until you have a need to use it else where and then follow the rules to make it either a module or base Algorithm.

In summary we are now trying very hard on the Mantid project to name anything as a *utility* *helper* or *manager*, as the clients of these tools have no idea what's in them based on the name and they end up becoming dumping grounds for unstructured code.

This is a rather long response. I'm happy to merge your code as is, if you like, but if you can clean up the mlzutilities in the near future as I have described above, I would be very grateful.